### PR TITLE
Update 3rdparty/mkldnn to v1.2.2

### PR DESCRIPTION
**Pull request by TaoLv**
## Description ##
Currently it's the release candidate for v1.2.0. Will change to formal release once it's published.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by TaoLv** [comment]
This is to upgrade the 3rdparty dependency to DNNL v1.2.1 and the changes are ready for review. Hope someone can help to retrigger the failing case. Thanks.

**Comment by leezu** [comment]
@TaoLv you can retrigger yourself.
1) Click Login
image

2) Click the circular button in the upper right corner
image

**Comment by TaoLv** [comment]
Thanks for reminding, @leezu. Yes, I just got the privilege for re-triggering CI jobs last week~

File: tests/cpp/operator/mkldnn_test.cc [hunk 1]
```diff
@@ -100,7 +100,7 @@ static void VerifyDefMem(const mkldnn::memory &mem) {
 
 TEST(MKLDNN_UTIL_FUNC, MemFormat) {
   // Check whether the number of format is correct.
-  CHECK_EQ(mkldnn_format_tag_last, 131);
+  CHECK_EQ(mkldnn_format_tag_last, 152);
```
Comment (ciyongch) [inline tests/cpp/operator/mkldnn_test.cc]: Where can we get this number? Or do we need to calculate this number manually when upgrading to a new version of DNNL?

Comment (TaoLv) [reply tests/cpp/operator/mkldnn_test.cc]: We need print this flag in cpp to get the value. This test was design to notify us if there are new formats added to dnnl so we need carefully check if these new formats break any integration code or the integration code need be improved to handle them.

**Comment by TaoLv** [comment]
I noticed that DNNL released the 1.2.2 patch release today. I'm going to update this PR to the latest release of the library.

**Comment by szha** [comment]
Note that master branch is now used for 2.x development. If this needs to be backported to 1.7, it needs to be done manually.

**Pull request merged**